sdk/tests/conformance2/extensions/webgl-shader-pixel-local-storage.html

```diff
@@ -0,0 +1,334 @@
+let MAX_COLOR_ATTACHMENTS_WITH_ACTIVE_PIXEL_LOCAL_STORAGE = null;
+let MAX_COMBINED_DRAW_BUFFERS_AND_PIXEL_LOCAL_STORAGE_PLANES = null;
+let MAX_COLOR_ATTACHMENTS = null;
+let MAX_DRAW_BUFFERS = null;
+
+function arraysEqual(a, b) {
+  if (typeof a !== typeof b)
+    return false;
+  if (a.length != b.length)
+    return false;
+  for (let i = 0; i < a.length; ++i) {
+    if (a[i] !== b[i])
+      return false;
+  }
+  return true;
+}
+
+function runTest() {
+  if (!gl) {
+    return function() {
+      testFailed("WebGL2 context does not exist");
+    }
+  }
+
+  // Check the behavior surrounding WEBGL_shader_pixel_local_storage being enabled.
+  checkExtensionNotSupportedWhenDisabled();
+  checkDependencyExtensionsEnabled(false);
+  pls = gl.getExtension("WEBGL_shader_pixel_local_storage");
+  wtu.runExtensionSupportedTest(gl, "WEBGL_shader_pixel_local_storage", pls != null);
+  if (!pls)
+    return;
+  checkDependencyExtensionsEnabled(true);
+
+  // Query and verify PLS implementation dependent limits.
+  MAX_PIXEL_LOCAL_STORAGE_PLANES =
+    gl.getParameter(pls.MAX_PIXEL_LOCAL_STORAGE_PLANES_WEBGL);
+  MAX_COLOR_ATTACHMENTS_WITH_ACTIVE_PIXEL_LOCAL_STORAGE =
+    gl.getParameter(pls.MAX_COLOR_ATTACHMENTS_WITH_ACTIVE_PIXEL_LOCAL_STORAGE_WEBGL);
+  MAX_COMBINED_DRAW_BUFFERS_AND_PIXEL_LOCAL_STORAGE_PLANES =
+    gl.getParameter(pls.MAX_COMBINED_DRAW_BUFFERS_AND_PIXEL_LOCAL_STORAGE_PLANES_WEBGL);
+  wtu.glErrorShouldBe(gl, gl.NONE, "Pixel local storage queries should be supported.");
+  MAX_COLOR_ATTACHMENTS = gl.getParameter(gl.MAX_COLOR_ATTACHMENTS);
+  MAX_DRAW_BUFFERS = gl.getParameter(gl.MAX_DRAW_BUFFERS);
+  checkImplementationDependentLimits();
+
+  checkInitialValues();
+  checkWebGLNonNormativeBehavior();
+
+  gl.disable(gl.DITHER);
+  checkRendering();
+}
+
+// Check that a context does not support WEBGL_shader_pixel_local_storage before it is enabled.
+function checkExtensionNotSupportedWhenDisabled() {
+  shouldBeNull("gl.getParameter(0x96E0 /*MAX_PIXEL_LOCAL_STORAGE_PLANES_WEBGL*/)");
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "parameter unknown without enabling the extension");
+  shouldBeNull(
+    "gl.getParameter(0x96E1 /*MAX_COLOR_ATTACHMENTS_WITH_ACTIVE_PIXEL_LOCAL_STORAGE_WEBGL*/)");
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "parameter unknown without enabling the extension");
+  shouldBeNull(
+    "gl.getParameter(0x96E2 /*MAX_COMBINED_DRAW_BUFFERS_AND_PIXEL_LOCAL_STORAGE_PLANES_WEBGL*/)");
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "parameter unknown without enabling the extension");
+  shouldBeNull(
+    "gl.getParameter(0x96E3 /*PIXEL_LOCAL_STORAGE_ACTIVE_PLANES_WEBGL*/)");
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "parameter unknown without enabling the extension");
+  wtu.glErrorShouldBe(gl, gl.NONE);
+}
+
+// Check that the dependency extensions of WEBGL_shader_pixel_local_storage are properly enabled or
+// disabled.
+function checkDependencyExtensionsEnabled(enabled) {
```

Comment:
As the first thing in this function could you log something like:

`debug("Checking that dependent extensions are " + (enabled ? "enabled" : "disabled"));`

Otherwise if errors crop up below, it'll be difficult to tell from the logs which case failed.

Reply:
Done

Reply:
Did you force-push the updated version here? I'm not seeing these changes.

Reply:
Derp. Forgot to commit my local changes.